### PR TITLE
doc: Add documentation for runner.md under api

### LINKED ISSUE #651: Add documentation for `anta.runner` module

### PR DESCRIPTION
**Pull request by gmuloc**
# Description

Fixes #651 

# Reviewer questions

- [x] Move scripts only in snippets?
    No - use examples
- [x] Moving more stuff around?
    Not for now
 
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have run pre-commit for code linting and typing (`pre-commit run`)
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes (`tox -e testenv`)

### CODE REVIEW

**Review by mtache — APPROVED** [review]

File: docs/scripts/anta_runner.py
Comment (titom73) [inline docs/scripts/anta_runner.py:1]: May be we could move this file under `examples/` fodler to be consistent with other examples maintained in the repository?

**Review by titom73 — APPROVED** [review]

File: docs/scripts/anta_runner.py
Comment (gmuloc) [reply docs/scripts/anta_runner.py:1]: YEah! the symlink for snippets would still be needed then :) but fine by me
will do this

Comment (titom73) [reply docs/scripts/anta_runner.py:1]: I moved the script and refreshed symlink. We can merge once you validate it is OK for you

Comment (gmuloc) [reply docs/scripts/anta_runner.py:1]: ok I removed symlink and added an extra location for snippets

**Pull request merged**